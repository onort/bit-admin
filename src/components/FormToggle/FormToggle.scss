@import "../../styles/main.scss";

$height: px-to-rem(28px);

.container {
  display: flex;
  align-items: center;
}

.switch {
  position: relative;
  display: inline-block;
  width: px-to-rem(48px);
  height: $height;
}

.switch input {
  opacity: 0;
  width: 0;
  height: 0;
}

.slider {
  position: absolute;
  cursor: pointer;
  border-radius: $height;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background-color: darken($light-bg, 20);
  transition: 0.4s;
}

.slider:before {
  position: absolute;
  border-radius: 50%;
  content: "";
  height: px-to-rem(20px);
  width: px-to-rem(20px);
  left: px-to-rem(4px);
  bottom: px-to-rem(4px);
  background-color: $white;
  transition: 0.4s;
}

input:checked {
  & + .slider {
    background-color: $green;
  }
}

input:checked {
  & + .slider:before {
    transform: translateX(20px);
  }
}

.label {
  display: inline-block;
  font-size: $font-size-md;
  font-weight: 600;
  letter-spacing: 0.66px;
  margin-bottom: $spacer-xxsm;
  padding-right: $spacer-md;
}
