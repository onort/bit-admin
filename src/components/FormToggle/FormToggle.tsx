import React from "react"
import { FieldProps } from "formik"

import styles from "./FormToggle.scss"

interface Props extends FieldProps {
  label: string
}

const FormToggle: React.SFC<Props> = ({ form, field, ...props }) => {
  return (
    <div className={styles.container}>
      <span className={styles.label}>{props.label}</span>
      <label className={styles.switch}>
        <input {...field} {...props} />
        <span className={styles.slider} />
      </label>
    </div>
  )
}

export default FormToggle
