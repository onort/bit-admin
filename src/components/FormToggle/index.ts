import FormToggle from "./FormToggle"

export default FormToggle
