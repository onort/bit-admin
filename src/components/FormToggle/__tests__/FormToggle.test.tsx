import React from "react"
import { mount, shallow } from "enzyme"
import { Formik, Form, Field } from "formik"

import FormToggle from "../"

describe("<FormToggle />", () => {
  it("should match snapshot", () => {
    const fakeSubmit = jest.fn()
    const wrapper = shallow(
      <Formik initialValues={{ test: false }} onSubmit={fakeSubmit}>
        <Form>
          <Field
            type="checkbox"
            name="test"
            label="Test"
            component={FormToggle}
          />
        </Form>
      </Formik>
    )
    expect(wrapper.html()).toMatchSnapshot()
  })

  it("should render a span element as label with correct text", () => {
    const fakeSubmit = jest.fn()
    const testLabel = "Test"
    const wrapper = mount(
      <Formik initialValues={{ test: false }} onSubmit={fakeSubmit}>
        <Form>
          <Field
            type="checkbox"
            name="test"
            label={testLabel}
            component={FormToggle}
          />
        </Form>
      </Formik>
    )
    expect(wrapper.find(".label").exists()).toBe(true)
    expect(wrapper.find(".label").text()).toBe(testLabel)
    wrapper.unmount()
  })

  // TODO: Check formValue change
  it.skip("should toggle when clicked", () => {
    const fakeSubmit = jest.fn()
    const testLabel = "Test"
    const wrapper = mount(
      <Formik initialValues={{ test: false }} onSubmit={fakeSubmit}>
        <Form>
          <Field
            type="checkbox"
            name="test"
            label={testLabel}
            component={FormToggle}
          />
        </Form>
      </Formik>
    )
    wrapper.find("input").simulate("click")
    expect(wrapper.find("input:checked").exists()).toBe(true)
    wrapper.unmount()
  })
})
