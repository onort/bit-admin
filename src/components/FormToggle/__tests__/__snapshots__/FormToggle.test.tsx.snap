// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormToggle /> should match snapshot 1`] = `"<form><div class=\\"container\\"><span class=\\"label\\">Test</span><label class=\\"switch\\"><input type=\\"checkbox\\" name=\\"test\\" label=\\"Test\\"/><span class=\\"slider\\"></span></label></div></form>"`;
